src/Psalm/Internal/PhpVisitor/Reflector/FunctionLikeDocblockParser.php

```diff
@@ -496,8 +498,11 @@ public static function parse(PhpParser\Comment\Doc $comment): FunctionDocblockCo
             $info->no_named_args = true;
         }
 
-        $info->ignore_nullable_return = isset($parsed_docblock->tags['psalm-ignore-nullable-return']);
-        $info->ignore_falsable_return = isset($parsed_docblock->tags['psalm-ignore-falsable-return']);
+        $config = Config::getInstance();
+        $info->ignore_nullable_return = isset($parsed_docblock->tags['psalm-ignore-nullable-return'])
+            && $config->ignore_internal_nullable_issues;
+        $info->ignore_falsable_return = isset($parsed_docblock->tags['psalm-ignore-falsable-return'])
+            && $config->ignore_internal_falsable_issues;
```

Comment:
I'm not sure it's the best place to have it. Could we perhaps move it to a method on function storage instead?

Reply:
Not sure what you make reference to?

Reply:
The parser generates in-memory representation of a docblock. It feels kinda wrong to vary it based on external factors other than the docblock itself.

There's already an ugly hack in `FunctionLikeNodeScanner` that sets parameter flags for functions coming from internal stubs:
https://github.com/vimeo/psalm/blob/85f2083acb7dcb09d636baae3912cdb6bb7e9c2d/src/Psalm/Internal/PhpVisitor/Reflector/FunctionLikeNodeScanner.php#L207-L213

Could we perhaps tidy it up and reuse? Scanner does have a reference to codebase so it can access config without resorting to `Config::getInstance()`.

Reply:
I listed all the internals stubs in an array on config, that way I can check where the stub is coming from and I don't even need the flag. WDYT?

Reply:
Now that looks much better!

Reply:
I'd be inclined to go one step further and not require the `@psalm-ignore-nullable-return` at all in the stub and adding it automatically where we see false or null coming from the stubs return type. I think that's already what's being done in Callmap.

It would prevent forgetting to add this tag everywhere in the stubs.

You think it's a good idea?

Reply:
Say we have a function in the stub that shouldn't have its nullable return ignored. How would we opt out of that automatic flagging?

Reply:
I don't remember we ever had to create a stub to avoid that with CallMap so I guess it doesn't happen. If it did anyway, we would have to make a list of those function somewhere and make it an exception to the code I'll do that will make that will detect null and false and put the flag when needed

Reply:
Ok

Reply:
Mmh, for CallMap it seems handled in FunctionCallReturnTypeFetcher and it's more subtle than I though. I'll stay like this for now